crates/extrinsics/src/call.rs

```diff
@@ -90,12 +87,123 @@ pub struct CallCommand {
     output_json: bool,
 }
 
+/// A builder for the call command.
+pub struct CallCommandBuilder<Message, ExtrinsicOptions> {
+    opts: CallCommand,
+    marker: PhantomData<fn() -> (Message, ExtrinsicOptions)>,
+}
+
+impl<E> CallCommandBuilder<Missing<state::Message>, E> {
+    /// Sets the name of the contract message to call.
+    pub fn message<T: Into<String>>(
+        self,
+        message: T,
+    ) -> CallCommandBuilder<state::Message, E> {
+        CallCommandBuilder {
+            opts: CallCommand {
+                message: message.into(),
+                ..self.opts
+            },
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<M> CallCommandBuilder<M, Missing<state::ExtrinsicOptions>> {
+    /// Sets the extrinsic operation.
+    pub fn extrinsic_opts(
+        self,
+        extrinsic_opts: ExtrinsicOpts,
+    ) -> CallCommandBuilder<M, state::ExtrinsicOptions> {
+        CallCommandBuilder {
+            opts: CallCommand {
+                extrinsic_opts,
+                ..self.opts
+            },
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<M, E> CallCommandBuilder<M, E> {
+    /// Sets the the address of the the contract to call.
+    pub fn contract(self, contract: <DefaultConfig as Config>::AccountId) -> Self {
+        let mut this = self;
+        this.opts.contract = contract;
+        this
+    }
+
+    /// Sets the arguments of the contract message to call.
+    pub fn args(self, args: Vec<String>) -> Self {
```

Comment:
A `Vec<T> where T: ToString` would be a more useful type for this argument

Reply:
Also it would be nice to be able to supply arguments directly, not having to encode them as strings first. But for CLIs it doesn't matter, because strings is what we get anyways. I think this would be a nice addition in the future.